File: src/webcore/model/abstractlistmodel.cpp
#include <sway/webcore/model/abstractlistmodel.h>

NAMESPACE_BEGIN(sway)
NAMESPACE_BEGIN(webcore)
NAMESPACE_BEGIN(model)

AbstractListModel::AbstractListModel()
	: _dataItems(emscripten::val::array()) {
	// Empty
}

AbstractListModel::~AbstractListModel() {
	// Empty
}

void AbstractListModel::setItem(int idx, const emscripten::val & value) {
	_dataItems.set(idx, value);
}

emscripten::val AbstractListModel::getItems() const {
	return _dataItems;
}

NAMESPACE_END(model)
NAMESPACE_END(webcore)
NAMESPACE_END(sway)

File: src/webcore/model/abstractlistmodel.ts
"use strict";

export class AbstractListModel {
	public _wasm_listmodel: any = null;

	constructor(context: any) {
		this._wasm_listmodel = new context.module.AbstractListModel();
	}

	public getItemAt(idx: number): any {
		return this._wasm_listmodel.getItems()[idx];
	}

	public getItems(): any {
		return this._wasm_listmodel.getItems();
	}

	public setItem(idx, value): void {
		this._wasm_listmodel.setItem(idx, value._wasm_model);
	}
};
